src/Compiler/Service/ServiceParseTreeWalk.fs

```diff
@@ -50,6 +49,8 @@ type SyntaxNode =
 
 type SyntaxVisitorPath = SyntaxNode list
 
+type Ast = SyntaxNode list
```

Comment:
Nit: I would go with `SyntaxNodes` here. When I read `Ast` I imagine a tree instead of a list.

Reply:
Hmm, yes, but in this case it can arguably be viewed as the (defective) base of something like a rose tree (since any given `SyntaxNode` can have $\\{ 0,...,n \\}$ child nodes).

That is, the functions on the `Ast` module that operate on forests of sibling nodes feel more like operations on "the AST" than on "lists of nodes" to me—after all, `Ast = SyntaxNode list = ParsedInput.Contents`, i.e., the actual "tree" in `ParsedInput` (as opposed to the file metadata). `ParsedInput.Contents` is normally a singleton list anyway, unless there are multiple namespaces declared in the same file.

`SyntaxNodes` as the name of the corresponding module would also have the disadvantage of existing alongside the `SyntaxNode` module (which will provide a different set of more node-focused operations).

If you feel strongly about this, I'd probably be inclined just to go back using `ParsedInput` directly instead of exposing the underlying `SyntaxNode`s, even though I still feel like that's likely to leave some potential composability on the table.